### ISSUE TITLE
favs problem

### ISSUE THREAD

**Issue by bulletbites** [andstatus/andstatus#410]
The icons are NOT working properly or are flaky.

I fav'd something in the webclient, then I opened up the app, synced the home TL and voila:

![favproblem1](https://cloud.githubusercontent.com/assets/17816155/17647189/bbbd7d3c-61e7-11e6-89e0-3b2daa296404.png)

The heart icon didn't update to its colored state but instead I get a repost in the TL taking up valuable screen real estate for indicating that I fav'd something (instead of the heart switching color). Get rid of the repost stuff in the TL and make the icons function properly would have my preference.

suggestion: if its possible you could add a function to the timeline making the posting of your actions in the TL optional:  
![exampletloption1](https://cloud.githubusercontent.com/assets/17816155/17647454/049088c0-61f1-11e6-963a-a7df5f44a362.png)

With the TL clutter removed by adding this feature you could focus on making sure the thumbs work flawlessly.

**Comment by yvolk** [comment]
Regarding "Username1 favorited something by Username2” - this is how GNU Social works now. This is an ugly hack, which should be improved somehow by GNU Social developers. I mean this a new issue/feature candidate for this list: https://git.gnu.io/gnu/gnu-social/issues
Unfortunately, it looks like it's visited by developers not very often...

**Comment by bulletbites** [comment]
I'm afraid the flakyness of the thumbs also happens when using the message menu and not the thumbs. If I press Favorite or Repost AndStatus has trouble actually reposting or fav-ing the posts. In many cases I keep on pressing the actions but it doesn't work. It's like it simply doesn't register the action. A few updates ago at least the fav/reposting thing by using the message menu was always working. Maybe this only happens on my phone, I'm not sure. 

@yvolk could you run some tests on this yourself? Does both the thumbs or the menu actions work as intended on your AndStatus?

**Comment by yvolk** [comment]
First of all, these operations are asynchronous, which mean that when you press ”heart” icon, some new command to be sent to a server is put into sync queue. Timeline is not changed yet.
Only after successful interaction with a server AndStatus receives a response from it, telling if the server actually have set the "favorited" flag... 
If yes, AndStatus refreshes a Timeline view, showing updated "favorited" status... but it may postpone  this refresh in a case some previous refresh occurred too recently.

But this is not all. GNU Social (it looks like. to be checked...) _never_ sets "favorited" flag for a message in the API. This is why AndStatus may think that the messais still not favorited and this is why it continues to show you the message as not favorited. I will try to solve this by finding ”favoriting actions" by parsing text of messages...

**Comment by bulletbites** [comment]
Actions still not working (fav, repost), any progress in getting this solved; It's a major functionality bug?

**Comment by yvolk** [comment]
Actions on pressing buttons below messages are fixed in AndStatus v.36. Please try https://github.com/andstatus/andstatus/issues/456
